File: tournaments/countWays/countWays.py
def countWays(n, k):
    MOD = 10 ** 9 + 7

    c_nk = []
    for i in range(n + 1):
        c_nk.append([])
        for j in range(k + 1):
            c_nk[i].append(0)
    c_nk[0][0] = 1
    for i in range(n + 1):
        for j in range(k + 1):
            if i + 1 <= n:
                c_nk[i + 1][j] = (c_nk[i][j] + c_nk[i + 1][j]) % MOD
                if j + 1 <= k:
                    c_nk[i + 1][j + 1] += c_nk[i][j]
                    c_nk[i + 1][j + 1] %= MOD

    return c_nk[n][k]

File: tournaments/countWays/countWays.cpp
int countWays(int n, int k) {
  const int MOD = int(1e9) + 7;

  std::vector<std::vector<int>> c_nk;
  for (int i = 0; i <= n; i++) {
    std::vector<int> line;
    c_nk.push_back(line);
    for (int j = 0; j <= k; j++) {
      c_nk[i].push_back(0);
    }
  }
  c_nk[0][0] = 1;
  for (int i = 0; i < n; i++) {
    for (int j = 0; j <= k; j++) {
      if (i + 1 <= n) {
        c_nk[i + 1][j] = (c_nk[i + 1][j] + c_nk[i][j]) % MOD;
        if (j + 1 <= k) {
          c_nk[i + 1][j + 1] = (c_nk[i + 1][j + 1] + c_nk[i][j]) % MOD;
        }
      }
    }
  }

  return c_nk[n][k];
}
